onnxruntime/core/xnnpack/conv.cc

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/xnnpack/conv.h"
+
+#include "core/common/safeint.h"
+#include "core/xnnpack/build_kernel_info.h"
+#include "core/xnnpack/schema/xnnpack_onnx_defs.h"
+#include "core/framework/tensorprotoutils.h"
+
+#define XNNPACK_CPU_MS_DOMAIN_OPERATOR_KERNEL(name, ver, builder, ...) \
+  ONNX_OPERATOR_KERNEL_EX(name, kXNNPackDomain, ver, kCpuExecutionProvider, builder, __VA_ARGS__)
+
+namespace onnxruntime {
+namespace xnnpack {
+
+static bool IsAllDimKnown(const TensorShape& s) {
+  size_t len = s.NumDimensions();
+  for (size_t i = 0; i != len; ++i) {
+    if (s[i] < 0) return false;
```

Comment:
>< [](http://example.com/codeflow?start=13&length=1)

<=

Reply:
I set it to < because I don't know if the current code can handle zero dim inputs.